code_examples/sdk_examples/src/workshop/attester/generateDid.ts

```diff
@@ -1,51 +1,40 @@
 import { config as envConfig } from 'dotenv'
 
 import * as Kilt from '@kiltprotocol/sdk-js'
-
+// TODO: No longer needed in content, but needed for tutorial?
+// TODO: What to do?
+// TODO: And generateAccount needed changes, why?
 import { generateAccount } from './generateAccount'
-import { generateKeypairs } from './generateKeypairs'
-
-export async function createFullDid(
-  submitterAccount: Kilt.KiltKeyringPair
-): Promise<{
-  mnemonic: string
-  fullDid: Kilt.DidDocument
-}> {
-  const api = Kilt.ConfigService.get('api')
 
-  const mnemonic = Kilt.Utils.Crypto.mnemonicGenerate()
-  const {
-    authentication,
-    keyAgreement,
-    assertionMethod,
-    capabilityDelegation
-  } = generateKeypairs(mnemonic)
-  // Get tx that will create the DID on chain and DID-URI that can be used to resolve the DID Document.
-  const fullDidCreationTx = await Kilt.Did.getStoreTx(
-    {
-      authentication: [authentication],
-      keyAgreement: [keyAgreement],
-      assertionMethod: [assertionMethod],
-      capabilityDelegation: [capabilityDelegation]
-    },
-    submitterAccount.address,
-    async ({ data }) => ({
-      signature: authentication.sign(data),
-      keyType: authentication.type
+export async function createFullDid() {
+  const api = await Kilt.connect(process.env.WSS_ADDRESS as string);
+  const accountMnemonic = process.env.ATTESTER_ACCOUNT_MNEMONIC as string;
```

Comment:
in principle this works, but these files are tested, right? creating a DID can only ever be done one, so we'd have to create a new ATTESTER_ACCOUNT_MNEMONIC and send funds to the account before each run?

Reply:
@rflechtner Yes, but we always used to create a DID anyway, so actually I am not sure how the tests ever worked. Couple of options to balance getting our tests to work and a tutorial people can follow…

1. Create a new random DID every time
2. Create a new account each time, to create a new DID each time etc
3. Wrap the code to only create a DID if it doesn't already exist

I think 3 is the simplest and tidiest, and we make sure the tutorial is clear. You need to plug your own data in, etc, maybe through the _.env_ file.

Reply:
1. Is what we used to do before, but is not an option because we now create the did based on an account.
Tweaked to make the "random did" approach work again gives us 2. -> make random account, send accounts tokens, create did from account. I think that's the best solution because if we follow 3 everywhere, we're not actually testing much, we'll just skip through all creation operations

Reply:
@rflechtner fair enough, i guess it's on test net anyway